lib/matroid.rb

```diff
@@ -1,43 +1,75 @@
-require "matroid/version"
-require 'httparty'
-require 'pry'
-
+require 'dotenv/load'
+require 'matroid/connection'
+require 'matroid/version'
+require 'matroid/detector'
 module Matroid
-  class MatroidClient
-    def initialize(options)
-      @base_url = options[:base_url]
-      @client_id = options[:client_id]
-      @client_secret = options[:client_secret]
-      @request = HTTParty
+  class << self
+
+    # Authenticates access for Matroid API
+    # @example
+    #  Matroid.authenticate("<your_client_id>", "<your_client_secret>")
+    # @param client_id [String]
+    # @param client_secret [String]
+    # @return [Boolean] If the the access token is successfully created.
+    def authenticate(client_id = nil, client_secret = nil)
+      return true unless @token.nil? or @token.expired?
```

Comment:
`||` is better than `or` in Ruby for boolean logic. `and` and `or` are more for control flow: http://www.virtuouscode.com/2014/08/26/how-to-use-rubys-english-andor-operators-without-going-nuts/